src/python/pants/testutil/engine/util.py

```diff
@@ -143,7 +145,13 @@ def configurations(self):
 TARGET_TABLE = SymbolTable({'struct': Struct, 'target': Target})
 
 
-def assert_equal_with_printing(test_case, expected, actual):
+def remove_whitespace_from_error_message(s: str) -> str:
```

Comment:
Does this apply to any error message or specifically graph error messages? If the former, we might want this to be in something like `strutil.py` or `TestBase`. I always struggle to know where to put things..

Reply:
I think this would be specifically graph error messages. I've since moved this method to `test_rules.py`, and made `assert_equal_with_printing()` accept a `uniform_formatter` argument, which formats the expected and actual text before comparing it. I think that `remove_whitespace_from_graph_output()` in `test_rules.py` is likely to be graph-specific, but further work will look to see if there's a useful more generic version to put in `TestBase` (I *think* that'd be the right place here?).